src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -75,316 +67,64 @@ public enum Flags
-            AddPunctuator(PunctuatorBang, TokKind.Bang);
-            AddPunctuator(PunctuatorAdd, TokKind.Add);
-            AddPunctuator(PunctuatorSub, TokKind.Sub);
-            AddPunctuator(PunctuatorMul, TokKind.Mul);
-            AddPunctuator(PunctuatorDiv, TokKind.Div);
-            AddPunctuator(PunctuatorCaret, TokKind.Caret);
-            AddPunctuator(PunctuatorParenOpen, TokKind.ParenOpen);
-            AddPunctuator(PunctuatorParenClose, TokKind.ParenClose);
-            AddPunctuator(PunctuatorEqual, TokKind.Equ);
-            AddPunctuator(PunctuatorLess, TokKind.Lss);
-            AddPunctuator(PunctuatorLessOrEqual, TokKind.LssEqu);
-            AddPunctuator(PunctuatorGreater, TokKind.Grt);
-            AddPunctuator(PunctuatorGreaterOrEqual, TokKind.GrtEqu);
-            AddPunctuator(PunctuatorNotEqual, TokKind.LssGrt);
-            AddPunctuator(LocalizedPunctuatorListSeparator, TokKind.Comma);
-            AddPunctuator(PunctuatorDot, TokKind.Dot);
-            AddPunctuator(PunctuatorColon, TokKind.Colon);
-            AddPunctuator(PunctuatorCurlyOpen, TokKind.CurlyOpen);
-            AddPunctuator(PunctuatorCurlyClose, TokKind.CurlyClose);
-            AddPunctuator(PunctuatorBracketOpen, TokKind.BracketOpen);
-            AddPunctuator(PunctuatorBracketClose, TokKind.BracketClose);
-            AddPunctuator(PunctuatorAmpersand, TokKind.Ampersand);
-            AddPunctuator(PunctuatorPercent, TokKind.PercentSign);
-            AddPunctuator(LocalizedPunctuatorChainingSeparator, TokKind.Semicolon);
-            AddPunctuator(PunctuatorAt, TokKind.At);
-
-            //Commenting punctuators
-            AddPunctuator(PunctuatorBlockComment, TokKind.Comment);
-            AddPunctuator(PunctuatorLineComment, TokKind.Comment);
-
-            // Keywords
-            AddKeyword(KeywordTrue, TokKind.True);
-            AddKeyword(KeywordFalse, TokKind.False);
-            AddKeyword(KeywordIn, TokKind.In);
-            AddKeyword(KeywordExactin, TokKind.Exactin);
-            AddKeyword(KeywordSelf, TokKind.Self);
-            AddKeyword(KeywordParent, TokKind.Parent);
-            AddKeyword(KeywordAnd, TokKind.KeyAnd);
-            AddKeyword(KeywordOr, TokKind.KeyOr);
-            AddKeyword(KeywordNot, TokKind.KeyNot);
-            AddKeyword(KeywordAs, TokKind.As);
-
-            PopulateKeywordArrays();
-        }
+        // Limits the StringBuilderCache TLS memory usage for LexerImpl.
+        // Usually our tokens are less than 128 characters long, unless it's a large string.
+        private const int DesiredStringBuilderSize = 128;
 
-        // This creates a new Lexer with the same locale as the original, which is a lot less expensive than spinning up a whole new one
-        private TexlLexer(TexlLexer original)
-        {
-            Contracts.AssertValue(original);
+        public static TexlLexer InvariantLexer { get; } = new TexlLexer(PunctuatorDecimalSeparatorInvariant);
```

Comment:
>{ get; } = new  [](http://example.com/codeflow?start=47&length=15)

As a property, this looks like it's returning an existing lexer; but each call to the property is actually creating a new lexer ....

Ie, it breaks this expectation: 
obj.ReferenceEquals(a.Prop1, a.Prop1);  


But also, creating a new lexer is expensive ... we still recreate the _punctuators dictionaries... that's state that could be shared across invokes.

Reply:
Discussed offline, confusion between `obj A { get; } = new ...` and `obj A {get { return new ... } }`
Will add a test covering the identity discussion.